locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/creator/DefaultAzureArmNetworkCreator.java

```diff
@@ -64,57 +68,71 @@ public static void createDefaultNetworkAndAddToTemplateOptionsIfRequired(Compute
+
         //Check if default already exists
-        Subnet preexistingSubnet = api.getSubnetApi(resourceGroupName, vnetName).get(subnetName);
+        Subnet preexistingSubnet = subnetApi.get(subnetName);
         if(preexistingSubnet != null){
             LOG.info("Using pre-existing default Azure network [{}] and subnet [{}] when provisioning machine", 
                     vnetName, subnetName);
             updateTemplateOptions(config, preexistingSubnet);
             return;
         }
 
+        createResourceGroupIfNeeded(api, resourceGroupName, location);
 
-        LOG.info("Network config not specified when provisioning Azure machine, and default network/subnet does not exists. "
-                + "Creating network [{}] and subnet [{}], and updating template options", vnetName, subnetName);
+        Subnet.SubnetProperties subnetProperties = Subnet.SubnetProperties.builder().addressPrefix(DEFAULT_SUBNET_ADDRESS_PREFIX).build();
 
-        createResourceGroupIfNeeded(api, resourceGroupName, location);
+        //Creating network + subnet if network doesn't exist
+        if(virtualNetworkApi.get(vnetName) == null) {
+            LOG.info("Network config not specified when provisioning Azure machine, and default network/subnet does not exists. "
+                    + "Creating network [{}] and subnet [{}], and updating template options", vnetName, subnetName);
 
-        //Setup properties for creating subnet/network
-        Subnet subnet = Subnet.create(subnetName, null, null,
-                Subnet.SubnetProperties.builder().addressPrefix(DEFAULT_SUBNET_ADDRESS_PREFIX).build());
+            Subnet subnet = Subnet.create(subnetName, null, null, subnetProperties);
 
-        VirtualNetwork.VirtualNetworkProperties virtualNetworkProperties = VirtualNetwork.VirtualNetworkProperties
-                .builder().addressSpace(VirtualNetwork.AddressSpace.create(Arrays.asList(DEFAULT_VNET_ADDRESS_PREFIX)))
-                .subnets(Arrays.asList(subnet)).build();
+            VirtualNetwork.VirtualNetworkProperties virtualNetworkProperties = VirtualNetwork.VirtualNetworkProperties
+                    .builder().addressSpace(VirtualNetwork.AddressSpace.create(Arrays.asList(DEFAULT_VNET_ADDRESS_PREFIX)))
+                    .subnets(Arrays.asList(subnet)).build();
+            virtualNetworkApi.createOrUpdate(vnetName, location, virtualNetworkProperties);
+        } else {
+            LOG.info("Network config not specified when provisioning Azure machine, and default subnet does not exists. "
+                    + "Creating subnet [{}] on network [{}], and updating template options", subnetName, vnetName);
 
-        //Create network
-        api.getVirtualNetworkApi(resourceGroupName).createOrUpdate(vnetName, location, virtualNetworkProperties);
+            //Just creating the subnet
+            subnetApi.createOrUpdate(subnetName, subnetProperties);
```

Comment:
Need to wait for the `createOrUpdate` operation to complete (here and above). The call is async. Can be done by polling the `properties.provisioningState()` returned by the `get` call on the resource.
If not done, any operations using the resource will fail.